test/composer.type.test.ts

```diff
@@ -110,7 +110,7 @@ describe("Composer types", () => {
                 const channelPostCaption = ctx.channelPost?.caption;
                 const channelPostText = ctx.channelPost?.text;
                 const match = ctx.match;
-                assertType<IsExact<typeof msgText, string>>(true);
+                assertType<IsExact<typeof msgText, string | undefined>>(true);
```

Comment:
Could you also assert that `ctx.msg.text ?? ctx.msg.caption` is `string`? If it isn't, then I'd like to fix up the filter queries first.

Reply:
It's `string | undefined`, no idea why :(

@carafelix #686 required touching 3 files, not 1 line, and uncovered a bug.

Reply:
It's odd to me that this seems to work correctly for `entities`→`text` but fails for `caption_entities`→`caption`. Smells like
- it's related to bad type distribution or 
- the above assumption is wrong

Reply:
`if (ctx.has(":caption_entities")) ctx.msg.caption` actually is `string`, as expected... Only `msg.text ?? msg.caption` breaks.

Reply:
Changing https://github.com/grammyjs/grammY/blob/a55168c7532be0761aa0ce54496ad7e0d25d68a5/src/temporary_types/message.ts#L66 to required "fixes" the issue

Reply:
> `if (ctx.has(":caption_entities")) ctx.msg.caption` actually is `string`, as expected... Only `msg.text ?? msg.caption` breaks.

Very interesting, that fits my intuition, but I'll still have to investigate it properly